tests/test_s3.py

```diff
@@ -3678,6 +3678,21 @@ def test_delete_bucket_encryption(self):
         with self.assertRaises(Exception):
             client.get_bucket_encryption(Bucket=bname)
 
+    def test_s3_invoke_lambda_assume_role_action(self):
+        session_factory = self.replay_flight_data("test_s3_invoke_lambda_assume_role")
```

Comment:
please patch the executor factory and augment table if you're going to use the s3 resource for testing. This reduces the amount of extra test data we record

```python
self.patch(s3.S3, "executor_factory", MainThreadExecutor)
self.patch(s3, "S3_AUGMENT_TABLE", [])
```